basic/payment-splitter/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+    struct PaymentSplitter {
+        /// This is the vault where all of the XRD that is to be splitted is stored
+        xrd_vault: Vault,
+
+        /// The resource definition of the admin badge. This admin badge is used to mint the NFTs which give the 
+        /// shareholders to authenticate them and keep track of how much they took so far from the splitter.
+        admin_badge_def: ResourceDef,
+
+        /// The vault containing the internal admin badge. This internal admin badge is mainly used for the updating
+        /// of NFT metadata and specifically in the `deposit_xrd` function when we want to update the quantity that
+        /// the shareholders are owed.
+        internal_admin_badge: Vault,
+
+        /// The resource definition of the shareholders NFT. This NFT is used to authenticate shareholders to allow
+        /// for the withdrawal of funds from the payment splitter and is also used to keep track of information about
+        /// this shareholder, the number of shares that they own, and the amount of XRD which they've been given so 
+        /// far.
+        shareholder_def: ResourceDef,
+
+        /// This decimal number is used to keep track of the number of shareholders that we have so far under this
+        /// payment splitter. This is used to allow for us to easily index all of the Shareholder NFTs.
+        number_of_shareholders: u128,
+    }
+
+    impl PaymentSplitter {
+        /// Creates a new Payment Splitter and returns it along with the admin badge back to the creator of the
+        /// payment splitter.
+        /// 
+        /// # Returns
+        /// 
+        /// * `Component` - A PaymentSplitter component
+        /// * `Bucket` - A bucket of the admin badge. This badge may be used to add shareholders to the PaymentSplitter.
+        pub fn new() -> (Component, Bucket) {
+            // Creating the admin badge
+            let admin_badge: Bucket = ResourceBuilder::new_fungible(DIVISIBILITY_NONE)
+                .metadata("name", "Admin Badge")
+                .metadata("symbol", "adm")
+                .initial_supply_fungible(1);
+
+            // Creating the internal admin badge
+            let internal_admin_badge: Bucket = ResourceBuilder::new_fungible(DIVISIBILITY_NONE)
+            .metadata("name", "Internal Admin Badge")
+            .metadata("symbol", "iadm")
+            .initial_supply_fungible(1);
+
+            // Creating the shareholders NFT Resource Definition. We will not mint any of them at this point
+            // of time as the minting and creation of these tokens happens when the `add_shareholder` function
+            // is called
+            let shareholder_def: ResourceDef = ResourceBuilder::new_non_fungible()
+                .metadata("name", "Shareholder")
+                .metadata("symbol", "shr")
+                .flags(MINTABLE | INDIVIDUAL_METADATA_MUTABLE | SHARED_METADATA_MUTABLE)
```

Comment:
You don't need the flag `SHARED_METADATA_MUTABLE` and the permission `MAY_CHANGE_SHARED_METADATA` since you are only mutating **individual** metadata in your example